src/utils.ts

```diff
@@ -53,32 +55,39 @@ export const extractSecret = (
 	core.setSecret(secretValue);
 };
 
+export const buildSecretResolver = (): SecretReferenceResolver => {
+	if (process.env[envServiceAccountToken]) {
+		return new Account();
+	} else {
+		return new Connect();
+	}
+};
+
 export const loadSecrets = async (shouldExportEnv: boolean): Promise<void> => {
-	// Pass User-Agent Information to the 1Password CLI
-	setClientInfo({
-		name: "1Password GitHub Action",
-		id: "GHA",
-		build: semverToInt(version),
-	});
-
-	// Load secrets from environment variables using 1Password CLI.
-	// Iterate over them to find 1Password references, extract the secret values,
-	// and make them available in the next steps either as step outputs or as environment variables.
-	const res = await exec.getExecOutput(`sh -c "op env ls"`);
-
-	if (res.stdout === "") {
+	const refs = loadSecretRefsFromEnv();
+
+	if (refs.length === 0) {
 		return;
 	}
 
-	const envs = res.stdout.replace(/\n+$/g, "").split(/\r?\n/);
-	for (const envName of envs) {
-		extractSecret(envName, shouldExportEnv);
+	const resolver = buildSecretResolver();
+
+	for (const key of refs) {
+		await extractSecret(resolver, key, shouldExportEnv);
 	}
+
 	if (shouldExportEnv) {
-		core.exportVariable(envManagedVariables, envs.join());
+		core.exportVariable(envManagedVariables, refs.join());
 	}
 };
 
+export const loadSecretRefsFromEnv = (): string[] => {
+	// secret references `op://<vault-name>/<item-name>/[section-name/]<field-name>`
+	return Object.entries(process.env)
+		.filter(([, v]) => v && v.startsWith("op://"))
+		.map(([k]) => k);
+};
+
```

Comment:
Great catch, I didn’t realize this was a breaking change. In fb0f77ad5b343f789583177e4109cadb7e6d6dee, I chose to introduce `⁠core.warning`[^1] to ensure compatibility with the previous version.

[^1]: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-error-message

Reply:
unit test in 1e6ff97b8bb23543ee46081b490675044968b87c

Reply:
Wondering if now that we have SDK, we can use [`validateSecretReference` function](https://github.com/1Password/onepassword-sdk-js/blob/86b10dd8a133c93f150cab0dd404ab8f6d6d0025/client/src/secrets.ts#L48) instead of implementing a separate matching/validation of secret references in this integration? Advantage would be that if secret reference syntax is expanded with additional concepts (e.g. like we've done when [adding `?attribute=otp` syntax](https://developer.1password.com/docs/cli/secret-reference-syntax#field-and-file-metadata-attributes)), those can be added to this integration with just a version bump of the SDK and don't require code changes in this integration itself.

Reply:
@SimonBarendse It sounds like a good idea, but I'm afraid we can't do it because the `src/auth/connect.ts` side is temporarily not compatible. If the validation here is successful but uses `connect`, it will result in an error.

Reply:
Another piece worth noting is that the code currently implemented for Connect will stay and we won't be able to fully replace it with [validateSecretReference function](https://github.com/1Password/onepassword-sdk-js/blob/86b10dd8a133c93f150cab0dd404ab8f6d6d0025/client/src/secrets.ts#L48) because that one doesn't give us the pieces of the reference: vault, item, (optional) section and field. We would still need to do that based on regex. 

Therefore I'm ok with keeping this as is and not intertwine SDK function in the Connect functionality.

Reply:
I will resolve this since now the functionality is now backwards compatible.